cmd/kcp-catalog/bind/catalogentry/cmd.go

```diff
@@ -0,0 +1,207 @@
+}
+
+// NewBindOptions returns new BindOptions.
+func NewBindOptions(streams genericclioptions.IOStreams) *BindOptions {
+	return &BindOptions{
+		Options: base.NewOptions(streams),
+	}
+}
+
+// BindFlags binds fields to cmd's flagset.
+func (b *BindOptions) BindFlags(cmd *cobra.Command) {
+	b.Options.BindFlags(cmd)
+}
+
+// Complete ensures all fields are initialized.
+func (b *BindOptions) Complete(args []string) error {
+	if err := b.Options.Complete(); err != nil {
+		return err
+	}
+
+	if len(args) > 0 {
+		b.CatalogEntryRef = args[0]
+	}
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (b *BindOptions) Validate() error {
+	if b.CatalogEntryRef == "" {
+		return errors.New("`root:ws:catalogentry_object` reference to bind is required as an argument")
+	}
+
+	if !strings.HasPrefix(b.CatalogEntryRef, "root") {
+		return fmt.Errorf("fully qualified reference to workspace where Catalog Entry exists is required. The format is `root:<ws>:<catalogentry>`")
+	}
+
+	return b.Options.Validate()
+}
+
+// Run creates an apibinding for the user.
+func (b *BindOptions) Run(ctx context.Context) error {
+	config, err := b.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	kcpclient, err := newKCPClusterClient(config)
+	if err != nil {
+		return err
+	}
+
+	_, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return err
+	}
+
+	// get the entry refered in the command.
+	path, entryName := logicalcluster.New(b.CatalogEntryRef).Split()
+
+	client, err := newCatalogClient(config)
+	if err != nil {
+		return err
+	}
+
+	// TODO: figure out if this usage of controller-runtime client can be used.
+	entry := catalogv1alpha1.CatalogEntry{}
+	// TODO: context shouldn't embed path directly, figure out if the host URL will be correct.
+	err = client.Get(logicalcluster.WithCluster(ctx, path), types.NamespacedName{Name: entryName}, &entry)
+	if err != nil {
+		return err
+	}
+
+	apibindings := []apisv1alpha1.APIBinding{}
+	for _, ref := range entry.Spec.Exports {
+		// check if ref is valid. Skip if invalid by logging error.
+		if ref.Workspace.Path == "" || ref.Workspace.ExportName == "" {
+			klog.Infof("invalid reference %q/%q", ref.Workspace.Path, ref.Workspace.ExportName)
+			continue
+		}
+
+		apibinding := &apisv1alpha1.APIBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ref.Workspace.ExportName,
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
+				Reference: ref,
+			},
+		}
+
+		apibindings = append(apibindings, *apibinding)
+	}
+
+	// Create bindings to the target workspace
+	for _, binding := range apibindings {
+		_, err := kcpclient.Cluster(currentClusterName).ApisV1alpha1().APIBindings().Create(ctx, &binding, metav1.CreateOptions{})
+		if err != nil {
+			// If an APIBinding already exists, intentionally not updating it since we would not like reset AcceptablePermissionClaims.
+			klog.Infof("Failed to create API binding %s: %w", binding.Name, err)
```

Comment:
Please use contextual logging instead of package-level klog statements (logger := klog.FromContext(ctx))